src/trees/MWNode.h

```diff
@@ -185,6 +188,8 @@ template <int D> class MWNode {
 
     void setMaxSquareNorm();
     void resetMaxSquareNorm();
+    double calcMaxSquareNorm() const { return std::pow(2.0, D * getScale()) * getSquareNorm(); }
```

Comment:
This should be renamed as something like calcScaledSquareNorm(). (It does not return the MaxSquareNorm unless it is a special node)

Reply:
Otherwise it is good for commit!

Reply:
done